core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -80,8 +80,8 @@ import org.apache.spark.util.kvstore._
  * break. Simple streaming of JSON-formatted events, as is implemented today, implicitly
  * maintains this invariant.
  */
-private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
-  extends ApplicationHistoryProvider with Logging {
+private[history] class FsHistoryProvider(conf: SparkConf, protected val clock: Clock)
+  extends ApplicationHistoryProvider with LogFilesBlacklisting with Logging {
```

Comment:
What is the special advantage of using a mixin trait rather than directly changing the code here in `FsHistoryProvider`?

Reply:
I just wanted to separate the blacklisting logic since `FsHistoryProvider` contains already a lot of code. So I just considered it more readable. If you prefer I can inline it.

Reply:
This seems not so necessary, let's inline this trait.